detectorSegmentations/src/GridDRcalo_k4geo.cpp

```diff
@@ -68,13 +70,64 @@ namespace DDSegmentation {
     auto rot = paramBase->GetRotationZYX(noPhi);
     auto translation = rot * localPos;
 
+    // we want the cell position to be the front end of the tower
+    // retrieve the fiber length
+    int row = y(cID);
+    int col = x(cID);
+    double fiberLen = paramBase->GetShortFibers(noEta).retrieveFiberLength(row, col);
+    // and vector to the front end
+    auto diff = paramBase->GetTowerPos(noPhi) - waferPos;
+    auto unitVec = diff / std::sqrt(diff.Mag2());
+
+    // total vector is sum of the vector to the wafer center + rotated local coordinate
+    // + the translation to the front end of the tower
+    auto total = translation + waferPos + unitVec * fiberLen;
+
+    // if LHS rotate by 180 deg w.r.t. X axis (on par to the DRconstructor)
+    if (!isRHS)
+      total = dd4hep::RotationX(M_PI) * total;
+
+    dd4hep::printout(dd4hep::VERBOSE, "GridDRcalo_k4geo::position",
+                     "Hit position of (isRHS, noEta, noPhi, x, y) = (%d, %d, %d, %d, %d) is (x, y, z) = (%f, %f, %f)",
+                     isRHS, noEta, noPhi, col, row, total.x(), total.y(), total.z());
+
+    return Vector3D(total.x(), total.y(), total.z());
+  }
+
+  // rear end position (used for the digitization)
+  Vector3D GridDRcalo_k4geo::sipmPosition(const CellID& cID) const {
+    int noEta = numEta(cID); // always positive since we replicated the RHS
+    int noPhi = numPhi(cID);
+    bool isRHS = IsRHS(cID);
+
+    // first get a vector to the center of the wafer (per tower)
+    DRparamBase_k4geo* paramBase = setParamBase(noEta);
+    auto waferPos = paramBase->GetSipmLayerPos(noPhi);
+
+    dd4hep::Position localPos = dd4hep::Position(0., 0., 0.);
+
+    // get local coordinate
+    if (IsSiPM(cID))
```

Comment:
why is this `if` needed?

Reply:
This comes from a historical reason - this function was designed to return the position of the tower (precisely "wafer" behind the tower) rather than the individual hit position when `IsSiPM` returns 0.

Reply:
Sure, I've just done it now